bokehjs/src/coffee/models/plots/plot.coffee

```diff
@@ -352,11 +352,16 @@ class PlotView extends Renderer.View
     @update_range(null)
 
   build_levels: () ->
+    renderer_models = @mget("renderers")
+    for tool_model in @mget("tools")
+      synthetic = tool_model.get("synthetic_renderers") ? []
+      renderer_models = renderer_models.concat(synthetic)
```

Comment:
I will be doing further work in this area, so we don't have to decide now (this is internal logic anyway).